internal/sdkprovider/service/flink/flink_application_deployment.go

```diff
@@ -117,6 +117,22 @@ func resourceFlinkApplicationDeploymentCreate(
 		return diag.Errorf("cannot create Flink Application Deployment: %v", err)
 	}
 
+	err = waitForStateContext(
+		ctx, client,
+		project, serviceName, applicationID, r.ID,
+		[]string{
+			"INITIALIZING",
+			"CREATED",
+		},
+		[]string{
+			"RUNNING",
+		},
+		d.Timeout(schema.TimeoutCreate),
```

Comment:
It might stuck for whole 20 minutes on RESTARTING status.

Reply:
Yes, I figured it out. Thank you. I will push the fix soon.